### PR TITLE
[SPARK-17625] [SQL] set expectedOutputAttributes when converting SimpleCatalogRelation to LogicalRelation

### PR DESCRIPTION
**Pull request by wzhfy**
## What changes were proposed in this pull request?

When converting `SimpleCatalogRelation` to `LogicalRelation`, we also need to set `expectedOutputAttributes`, then the output in `LogicalRelation` is the attributes in `expectedOutputAttributes`, otherwise, it will use its schema to generate new attributes (and new exprId's), so the output of LogicalRelation will be different from output of SimpleCatalogRelation.

As a result, if we have a table in `InMemoryCatalog`, we can't lookup it (get a `SimpleCatalogRelation`) and then use it to compose `LogicalPlan`s.
## How was this patch tested?

add a test case

### CODE REVIEW

**Comment by wzhfy** [comment]
@cloud-fan @gatorsmile

**Comment by wzhfy** [comment]
May be the position of the test case is not right, where should I put it?

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala [hunk 1]
```diff
@@ -2310,6 +2308,18 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
...
+    val table = "tbl"
+    withTable("tbl") {
+      sql("CREATE TABLE tbl(i INT, j STRING) USING parquet")
+      val tableIdent = spark.sessionState.sqlParser.parseTableIdentifier(table)
+      val relation = spark.sessionState.catalog.lookupRelation(tableIdent)
+      val expr = relation.resolve("i")
+      val plan = Dataset.ofRows(spark, Project(Seq(expr), relation))
+      plan.queryExecution.assertAnalyzed()
+    }
+  }
```
Comment (gatorsmile) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala]: Your test case can be simplified to
```Scala
    val tableName = "tbl"
    withTable(tableName) {
      spark.range(10).select('id as 'i, 'id as 'j).write.saveAsTable(tableName)
      val relation = spark.sessionState.catalog.lookupRelation(TableIdentifier(tableName))
      val expr = relation.resolve("i")
      Dataset.ofRows(spark, Project(Seq(expr), relation))
    }
```

**Comment by gatorsmile** [comment]
The fix looks ok to me, but you need to improve your PR description. It might not be easy for reviewers to understand your fix based on your PR description. 

I do not know which test suite is the best place, but `SQLQuerySuite` is not good for sure. Your test case is not related to SQL. Let @cloud-fan make the decision. : )

**Comment by cloud-fan** [comment]
Let's put it in `DataFrameSuite`

**Comment by wzhfy** [comment]
@cloud-fan Ok.

**Comment by gatorsmile** [comment]
LGTM pending Jenkins

**Comment by cloud-fan** [comment]
LGTM

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**